upstairs/src/lib.rs

```diff
@@ -1602,7 +1627,54 @@ impl Downstairs {
                 }
             } else {
                 // no upstairs encryption context
-                responses
+                if let Ok(responses) = responses {
```

Comment:
It seems like there is a fair amount of the `if` above repeated here, but I don't see an easy way to combine the two branches.  We will now always have the hash to verify and it would be nice to always do that in the same code path, but as the encryption is optional, yeah, it's tricky.

Reply:
Tried to address in 901c587